dbms/src/DataStreams/WindowBlockInputStream.cpp

```diff
@@ -404,13 +444,18 @@ void WindowTransformAction::advanceFrameStart()
     }
     case WindowFrame::BoundaryType::Offset:
         if (window_description.frame.type == WindowFrame::FrameType::Rows)
-            frame_start = stepToFrameStart(current_row, window_description.frame);
+        {
+            RowNumber frame_start_tmp;
+            std::tie(frame_start_tmp, frame_started) = stepToFrameStart(current_row, window_description.frame);
+            if (frame_started)
+                frame_start = frame_start_tmp;
```

Comment:
why not
```suggestion
            std::tie(frame_start, frame_started) = stepToFrameStart(current_row, window_description.frame);
```
?
Does the value of `frame_start` make sense when `frame_started` is false? If it doesn't make sense, why not assign it directly?

Reply:
> why not
> 
> ? Does the value of `frame_start` make sense when `frame_started` is false? If it doesn't make sense, why not assign it directly?

When `frame_started` is false, frame_start should not be changed, or it will be less than `prev_frame_start`, this is invalid.